Chapter6/super_process.erl

```diff
@@ -0,0 +1,107 @@
+-module(super_process).
+-export([start_link/2, stop/1, add_child/2, stop_child/2]).
+-export([init/1]).
+
+% Interface
+start_link(Name, ChildSpecList) ->
+	register(Name, spawn_link(super_process, init, [ChildSpecList])), ok.
+
+stop(Name) ->
+	Name ! {stop, self()},
+	receive {reply, Reply} -> Reply end.
+
+
+% Exercise 6-3: A Supervisor Process (start_child & stop_child)
+add_child(Name, Process) ->
+	Name ! {add, self(), Process},
+	receive {reply, Reply} -> Reply end.
+
+stop_child(Name, Pid) ->
```

Comment:
ohhhh its because if a process crashes then it gets restarted but we don't know what the new restarted Pid is, I'll update the code now